File: PTA 1012 数字分类.cpp
/* PTA 1012 数字分类
 * 2022/07/29
 * */
#include <iostream>
#include <valarray>
using namespace std;

int main()
{
	int N;
	cin >> N;
	
	int A[6] = {0};
	int j = 0, count = 0;
	bool flag = false;
	while (N--)
	{
		int temp;
		cin >> temp;
		int temp2 = temp % 5;
		switch (temp2)
		{
			case 0:
				if (temp % 2 == 0)
					A[1] += temp;
				break;
			case 1:  //交错求和结果可能为零或根本没有数据, 要分这两种情况讨论, 这一点比较坑
				j++;
				flag = true;
				A[2] += int(pow(-1, j - 1)) * temp;
				break;
			case 2:
				A[3]++;
				break;
			case 3:
				A[4] += temp;
				count++;
				break;
			case 4:
				A[5] = max(A[5], temp);
				break;
			default:
				break;
		}
	}
	double ave;
	if (count != 0)
		ave = double(A[4]) / count;
	for (int i = 1; i <= 5; i++)
	{
		if (i == 2)
		{
			if (flag)  //表示有这个数字
				cout << A[i];
			else
				cout << "N";
		}
		else if (i == 4)
		{
			if (A[i] == 0)
				cout << "N";
			else
				printf("%.1f", ave);
		}
		else
		{
			if (A[i] == 0)
				cout << "N";
			else
				cout << A[i];
		}
		if (i != 5)
			cout << " ";
	}
}

File: PTA 1012 数字分类.java
/// PTA 1012 数字分类
/// 2023/01/06

import java.util.Scanner;

public class Main {
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int n = input.nextInt();
        int[] a = new int[5];
        int j = 0, count = 0, temp;
        while (n-- > 0) {
            temp = input.nextInt();
            switch (temp % 5) {
                case 0:
                    if (temp % 2 == 0)
                        a[0] += temp;
                    break;
                case 1:
                    if (j % 2 == 0)
                        a[1] = a[1] + temp;
                    else
                        a[1] = a[1] - temp;
                    j++;
                    break;
                case 2:
                    a[2]++;
                    break;
                case 3:
                    a[3] += temp;
                    count++;
                    break;
                case 4:
                    a[4] = Math.max(a[4], temp);
                    break;
                default:
                    break;
            }
        }
        double ave = 0;
        if (count != 0)
            ave = a[3] * 1.0 / count;
        for (int i = 0; i < 5; i++) {
            switch (i) {
                case 1:
                    if (j != 0)
                        System.out.print(a[i]);
                    else
                        System.out.print("N");
                    break;
                case 3:
                    if (a[i] == 0)
                        System.out.print("N");
                    else
                        System.out.printf("%.1f", ave);
                    break;
                default:
                    if (a[i] == 0)
                        System.out.print("N");
                    else
                        System.out.print(a[i]);
            }
            if (i != 4)
                System.out.print(" ");
        }
    }
}
